### ISSUE TITLE
Assertion failure in libuv

### ISSUE THREAD

**Issue by staticfloat** [JuliaLang/julia#7840]
I came across this assertion failure at the tail end of a `make testall` run.  @Keno any ideas?  I don't really know what's going on there in `uv__io_stop()`.

```
        From worker 4:       * repl
        From worker 2:       * test
        From worker 5:       * examples
        From worker 2:       * goto
        From worker 2:       * unicode
        From worker 2:       * parallel
Assertion failed: (loop->watchers[w->fd] == w), function uv__io_stop, file src/unix/core.c, line 764.

signal (6): Abort trap: 6
__pthread_kill at /usr/lib/system/libsystem_kernel.dylib (unknown line)
    SUCCESS
```

**Comment by tkelman** [comment]
Got another one of these on Ubuntu 14.04. Backtrace looks almost identical.

**Comment by tkelman** [comment]
closed by #12237? I'll keep an eye out for it and reopen if not
